### PR TITLE
abf interface improvements and tutorial

### LINKED ISSUE #72: [Documentation]: ABF conversion in gallery
### What would you like changed or added to the documentation and why?

ABF conversion in gallery

### Do you have any interest in helping write or edit the documentation?

No.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/nwb-conversion-tools/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/nwb-conversion-tools/issues)?

### PR DESCRIPTION
**Pull request by luiztauffer**
## Motivation

- [x] Some clean up on ABF data interface
- [x] fix #72 

## How to test the behavior?
```
from neuroconv import NWBConverter, AbfInterface

# Define the Data Interfaces for your converter
class MyIcephysConverter(NWBConverter):
    data_interface_classes = dict(AbfInterface=AbfInterface)

# Metadata info
icephys_metadata = {
    "cell_id": "20220512001",
    "slice_id": "20220512001",
    "targeted_layer": "L2-3(medial)",
    "inferred_layer": "",
    "recording_sessions": [
        {
            "abf_file_name": "2022_05_12_0000.abf",
            "stimulus_type": "Long_Square_currentClamp_-50 to100_step10pA",
            "icephys_experiment_type": "current_clamp"
        }
    ]
}

# Source data
source_data = dict(
    AbfInterface=dict(
        file_paths=["/media/luiz/storage/taufferconsulting/client_ben/project_icephys/22Q2_Ephys_DANDI/20220512001/2022_05_12_0000.abf"],
        icephys_metadata=icephys_metadata
    )
)

# Initialize converter
converter = MyIcephysConverter(source_data=source_data)

# Get metadata from source data and modify any values you want
metadata = converter.get_metadata()
metadata['NWBFile']['identifier'] = "ID12345"
metadata['NWBFile']['session_description'] = "Intracellular electrophysiology experiment."
metadata['NWBFile']['lab'] = "My lab name"
metadata['NWBFile']['institution'] = "My University"
metadata['NWBFile']['experimenter'] = ["John Doe", "Jane Doe"]
metadata["Subject"] = dict(
    subject_id="subject_ID123",
    species="Mus musculus",
    sex="M",
    date_of_birth="2022-03-15T00:00:00"
)

# Run conversion
converter.run_conversion(metadata=metadata, nwbfile_path='converted_icephys.nwb')
```

## Checklist

- [x] Have you thoroughly read our [Developer Guide](https://neuroconv.readthedocs.io/en/main/developer_guide.html)?
- [x] Have you ensured the PR description clearly describes the problem and solutions?
- [x] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/neuroconv/pulls) for the same change?
- [x] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?
- [x] Did you update the [CHANGLOG.md](https://github.com/catalystneuro/neuroconv/blob/main/CHANGELOG.md) file on your branch to describe your changes?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
 * Added package for viewing source code in the neuroconv documentation [PR #62](https://github.com/catalystneuro/neuroconv/pull/62)
 * Added Contributing guide for the Developer section of readthedocs. [PR #73](https://github.com/catalystneuro/neuroconv/pull/73)
 * Added style guide to the readthedocs [PR #28](https://github.com/catalystneuro/neuroconv/pull/28)
+* Added ABF data conversion tutorial [PR #89](https://github.com/catalystneuro/neuroconv/pull/89)
```
Comment (bendichter) [inline CHANGELOG.md:24]:
```suggestion
* Added ABF data conversion tutorial @luiztauffer [PR #89](https://github.com/catalystneuro/neuroconv/pull/89)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -43,6 +44,8 @@
 ### Improvements
 * Added function to add ImagingPlane objects to an nwbfile in `roiextractors` and corresponding unit tests. [PR #19](https://github.com/catalystneuro/neuroconv/pull/19)
 * Added function to add summary images from a `SegmentationExtractor` object to an nwbfile in the roiextractors module and corresponding unit tests [PR #22](https://github.com/catalystneuro/neuroconv/pull/22)
+* Small improvements on ABFInterface [PR #89](https://github.com/catalystneuro/neuroconv/pull/89)
```
Comment (bendichter) [inline CHANGELOG.md:47]:
```suggestion
* Small improvements on ABFInterface @luiztauffer [PR #89](https://github.com/catalystneuro/neuroconv/pull/89)
```

File: docs/conversion_examples_gallery/recording/abf.rst [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+Convert ABF intracellular electrophysiology data to NWB using :py:class:`~neuroconv.datainterfaces.icephys.abf.abfdatainterface.AbfInterface`.
+
+.. code-block:: python
+
+    >>> from neuroconv import NWBConverter, AbfInterface
+
+    >>> # Define the Data Interfaces for your converter
+    >>> class MyIcephysConverter(NWBConverter):
+    >>>     data_interface_classes = dict(AbfInterface=AbfInterface)
```
Comment (bendichter) [inline docs/conversion_examples_gallery/recording/abf.rst:12]: Could you make this work without an `NWBConverter`, just using the AbfInterface, like the other conversion examples?

File: docs/conversion_examples_gallery/recording/abf.rst [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    >>> metadata['NWBFile']['session_description'] = "Intracellular electrophysiology experiment."
+    >>> metadata['NWBFile']['lab'] = "My lab name"
+    >>> metadata['NWBFile']['institution'] = "My University"
+    >>> metadata['NWBFile']['experimenter'] = ["John Doe", "Jane Doe"]
+    >>> metadata["Subject"] = dict(
+    >>>     subject_id="subject_ID123",
+    >>>     species="Mus musculus",
+    >>>     sex="M",
+    >>>     date_of_birth="2022-03-15T00:00:00"
+    >>> )
```
Comment (bendichter) [inline docs/conversion_examples_gallery/recording/abf.rst:52]: Can you use `.update` and combine data together, e.g.

```python
metadata['NWBFile'].update(
    identifier="ID1234",
    session_description="Intracellular...",
    lab="my lab name"    # <-- optional
)

Comment (luiztauffer) [reply docs/conversion_examples_gallery/recording/abf.rst]: done

File: docs/conversion_examples_gallery/recording/abf.rst (hunk 3 above)
Comment (luiztauffer) [reply docs/conversion_examples_gallery/recording/abf.rst]: done

**Review by bendichter — APPROVED** [review]

**Review by bendichter — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
Since this was merged, I've started seeing a left-over file called 'converted_icephys.nwb' every time I run pytest.

@h-mayorquin Do you know what needs to be done for doctest to 'clean up' the files?

**Comment by luiztauffer** [comment]
@CodyCBakerPhD interesting, shouldn't this be happening with the other tutorials as well?
For the git actions I suppose it wouldn't be a problem, since the disk space for actions should be ephemeral. So it wouldn't make a difference to clean them up as an actions workflow step.

**Comment by CodyCBakerPhD** [comment]
> For the git actions I suppose it wouldn't be a problem, since the disk space for actions should be ephemeral. So it wouldn't make a difference to clean them up as an actions workflow step.

I mean for local testing, would rather not have unneeded files popping up and laying around in the directory.

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD @luiztauffer 
The other examples in the gallery do this by using fixtures:

To be specific, in the following line:
```
>>> nwbfile_path = f"{path_to_save_nwbfile}"
>>> interface.run_conversion(nwbfile_path=nwbfile_path, metadata=metadata)
```

The `path_to_save_nwbfile` is a fixture pointing to a temporary file defined in the `conftest.py` file:

https://github.com/catalystneuro/neuroconv/blob/ca427aa5994b742c2d3cdddd14212aa3a0758cb0/docs/conversion_examples_gallery/conftest.py#L12

Related to the discussion here:
https://github.com/catalystneuro/neuroconv/pull/24#discussion_r934181403

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin Yup I figured that out lol: https://github.com/catalystneuro/neuroconv/pull/99

**Pull request merged**